ArduCopter/mode.h

```diff
@@ -1164,14 +1194,31 @@ class ModeGuided : public Mode {
     void set_yaw_state(bool use_yaw, float yaw_cd, bool use_yaw_rate, float yaw_rate_cds, bool relative_angle);
 
     // controls which controller is run (pos or vel):
-    SubMode guided_mode = SubMode::TakeOff;
-    bool send_notification;     // used to send one time notification to ground station
-    bool takeoff_complete;      // true once takeoff has completed (used to trigger retracting of landing gear)
+    static SubMode guided_mode;
```

Comment:
could these cause problems? it makes it shared between all modes

Reply:
Feels like a bear trap!

Reply:
The problem the vast majority of the guided state is static already. This stuff: https://github.com/ArduPilot/ardupilot/blob/25e77a5d9fe360e178c86b386a4d2def90976dc4/ArduCopter/mode_guided.cpp#L9-L33

It either has to be all static or all not. With half and half as it is now you can endup being in the wrong sub mode for the targets you have. Making these static is a smaller change than making the existing stuff not.

Currently there should be no issue because there is nothing in the mode exit function so there is no overlap between one version of guided and the next. So long as everything is reset correctly in the mode init then we start clean (which I think it is already).

As you say, this is a bit nasty for the future, but anything else would require a much larger re-work.